--- a/src/libprojectM/Renderer/ShaderEngine.cpp
+++ b/src/libprojectM/Renderer/ShaderEngine.cpp
@@ -1226,7 +1226,7 @@ void ShaderEngine::reset()
 
 GLuint ShaderEngine::CompileShaderProgram(const std::string & VertexShaderCode, const std::string & FragmentShaderCode, const std::string & shaderTypeString){
 
-#if defined(WIN32) && !defined(EYETUNE_WINRT)
+#if defined(WIN32) && !defined(EYETUNE_WINRT) && !defined(HAS_ANGLE)
 	GLenum err = glewInit();
 #endif /** WIN32 */
 
--- a/src/libprojectM/projectM-opengl.h
+++ b/src/libprojectM/projectM-opengl.h
@@ -32,8 +32,12 @@
 #elif defined(_WIN32)
 #define GLM_FORCE_CXX03
 # include <windows.h>
+#if defined(HAS_ANGLE)
+#include <angle_gl.h>
+#else
 #include "glew.h"
 #include "wglew.h"
+#endif
 #else /* linux/unix/other */
 # ifdef USE_GLES
 #  include <GLES3/gl3.h>
