From 5e394b992d95e5a8ca4ba7aa3c3b91a84b8ce839 Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Wed, 20 May 2026 10:57:05 +0200
Subject: [PATCH] change dirent.h install path

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8bc6205..7b6667b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,7 +74,7 @@ install(
   FILES
     include/dirent.h
   DESTINATION
-    include/dirent-${DIRENT_VERSION}
+    include
   COMPONENT
     dev
 )
-- 
2.53.0

