From 8aafecc758c2864ee507bc4e83b8cc0f728c055f Mon Sep 17 00:00:00 2001
From: Arne Morten Kvarving <arne.morten.kvarving@sintef.no>
Date: Sun, 1 Oct 2017 21:13:14 +0200
Subject: [PATCH 4/6] fix namespace issue

---
 src/libprojectM/Common.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libprojectM/Common.hpp b/src/libprojectM/Common.hpp
index b292495..e8773b6 100755
--- a/src/libprojectM/Common.hpp
+++ b/src/libprojectM/Common.hpp
@@ -63,7 +63,7 @@ extern FILE *fmemopen(void *buf, size_t len, const char *pMode);
 
 #ifdef LINUX
 #include <cstdlib>
-#define projectM_isnan isnan
+#define projectM_isnan std::isnan
 
 #endif
 
-- 
2.7.4

