--- a/src/NativePresets/CMakeLists.txt
+++ b/src/NativePresets/CMakeLists.txt
@@ -25,11 +25,11 @@
 if(COMMAND cmake_policy)
        cmake_policy(SET CMP0003 NEW)
 endif(COMMAND cmake_policy)
-ADD_LIBRARY(RovastarDriftingChaos SHARED RovastarDriftingChaos.cpp)
-ADD_LIBRARY(RovastarDarkSecret SHARED RovastarDarkSecret.cpp)
-ADD_LIBRARY(RovastarFractalSpiral SHARED RovastarFractalSpiral.cpp)
-ADD_LIBRARY(RovastarFractopiaFrantic SHARED RovastarFractopiaFrantic.cpp)
-ADD_LIBRARY(MstressJuppyDancer SHARED MstressJuppyDancer.cpp)
+ADD_LIBRARY(RovastarDriftingChaos RovastarDriftingChaos.cpp)
+ADD_LIBRARY(RovastarDarkSecret RovastarDarkSecret.cpp)
+ADD_LIBRARY(RovastarFractalSpiral RovastarFractalSpiral.cpp)
+ADD_LIBRARY(RovastarFractopiaFrantic RovastarFractopiaFrantic.cpp)
+ADD_LIBRARY(MstressJuppyDancer MstressJuppyDancer.cpp)
 
 # Doesn't build yet
 #ADD_LIBRARY(RLGFractalDrop7c SHARED RLGFractalDrop7c.cpp)
