--- a/src/libprojectM/Renderer/Renderer.cpp
+++ b/src/libprojectM/Renderer/Renderer.cpp
@@ -267,11 +267,13 @@
     // when not 'renderToTexture', the user may use its own couple FBO/texture
     // so retrieve this external FBO if it exists, (0 means no FBO) and unbind it
     GLint externalFBO = 0;
+    #if !defined(__APPLE__)
     if (!renderTarget->renderToTexture)
     {
         glGetIntegerv(GL_FRAMEBUFFER_BINDING, &externalFBO);
         glBindFramebufferEXT(GL_FRAMEBUFFER_EXT, 0);
     }
+    #endif
 #endif
 
 	SetupPass1(pipeline, pipelineContext);
