From dc764c765c61dcc8feb91834baa2c9ea7bd232d0 Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Sun, 24 May 2026 00:31:59 +0200
Subject: [PATCH] Hack fix about macOS, as add-on link GL libs does it work

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index a62e8df..dd0f9fe 100644
--- a/configure.ac
+++ b/configure.ac
@@ -37,7 +37,7 @@ AS_IF([test "x$enable_emscripten" = "xyes" || test "x$EMSCRIPTEN" = "xyes"], [
   enable_sdl=yes
 ], [
   dnl Running in a normal OS (not emscripten)
-  AX_CHECK_GL
+ # AX_CHECK_GL
 
   # check OS if mac or linux
   AC_CANONICAL_HOST
-- 
2.39.5 (Apple Git-154)

